src/test/test-env-util.c

```diff
@@ -44,6 +44,16 @@ static void test_strv_env_get(void) {
         assert_se(streq(strv_env_get(l, "FOUR"), "4"));
 }
 
+static void test_strv_env_pairs_get(void) {
+        log_info("/* %s */", __func__);
+
+        char **l = STRV_MAKE("ONE_OR_TWO", "1", "THREE", "3", "ONE_OR_TWO", "2", "FOUR", "4");
```

Comment:
to catch the issue i pointed out above, maybe addd:

```
… "FIVE", "5", "SIX", "FIVE", "SEVEN", "7", …
```

and then query for "FIVE", and see if that returns "SEVEN" (which would incorrect), or "5" (which would be correct)